azure/services/virtualnetworks/virtualnetworks.go

```diff
@@ -91,6 +92,19 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			prefixes = to.StringSlice(existingVnet.VirtualNetworkPropertiesFormat.AddressSpace.AddressPrefixes)
 		}
 		vnet.CIDRBlocks = prefixes
+
+		// Update the subnet CIDRs if they already exist.
+		// This makes sure the subnet CIDRs are up to date and there are no validation errors when updating the VNet.
+		// Subnets that are not part of this cluster spec are silently ignored.
+		if existingVnet.Subnets != nil {
+			for _, subnet := range *existingVnet.Subnets {
+				if subnet.SubnetPropertiesFormat != nil && subnet.SubnetPropertiesFormat.AddressPrefix != nil {
```

Comment:
We have a similar block in the subnet service where we have to handle the single and multiple CIDR cases. What do you think about adding a converter or other helper function that reads in the SubnetPropertiesFormat and returns a list of addresses?